### PR TITLE
JAMES-2246 Integration tests for SMTP authentication + network matching

### PR DESCRIPTION
**Pull request by chibenwa**
Please review only 4 last commits.

Reopen of https://github.com/linagora/james-project/pull/1168

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

Stopped by error on the CI

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/NetworkMatcherIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+
+            messageSender.sendMessage(FROM, FROM);
+
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+
+            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/NetworkMatcherIntegrationTest.java]: Why do we wait for a fixed time ?

Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/NetworkMatcherIntegrationTest.java]: How do you ensure something did not happen?

I mean, you have to fix a duration after which you consider each environment would have performed the action, if it did it.

Yes, waiting for a side effect to happen is not clean, but :
 - i can not think about other strategies
 - we have always done like this.

However I have an alternative, but you won't like it, as it is less explicit in a test: define an alternative action, and wait for it.

Any opinion?

Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/NetworkMatcherIntegrationTest.java]: Tommorrow I will instead wait on a **ToMailRepository** mailet condition.

This involves having a probe for accessing **MailRepository**.

File: server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class MailRepositoryProbeImpl implements GuiceProbe {
+
+    private final MailRepositoryStore repositoryStore;
+
+    @Inject
+    public MailRepositoryProbeImpl(MailRepositoryStore repositoryStore) {
+        this.repositoryStore = repositoryStore;
+    }
+
+    public int getRepositrySize(String url) throws Exception {
```
Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java]: s/Repositry/Repository/

Comment (mbaechler) [inline server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java]: I think it deserves some javadoc : what is "size" for a mailrepository ?

Comment (chibenwa) [reply server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java]: The count of email?

File: server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java [hunk 3]
```diff
@@ -34,6 +34,9 @@ public MailRepositoryProbeImpl(MailRepositoryStore repositoryStore) {
         this.repositoryStore = repositoryStore;
     }
 
+    /**
+     * Get the count of email currently stored in a given repository
```
Comment (mbaechler) [inline server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java]: now, the method name doesn't match anymore. what about getRepositoryMailCount ?

Comment (strokyl) [reply server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java]: :+1:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**